src/main/java/api/Controller.java

```diff
@@ -126,46 +130,51 @@ public Hello hello(@RequestParam(value = "name", defaultValue = "team3") String
     public Movie movie(@PathVariable("title") String title,
                        @RequestParam(value = "year", defaultValue = "none") String year)
     {
-        ArrayList<Movie> matches = new ArrayList<>();
-        ArrayList<Movie> s = new ArrayList<>();
+        ArrayList<Movie> matchList = new ArrayList<>();
         Movie m = null;
-        String s1= "\\b(?i)("+title+")\\b"; // regex for searching the word start first in the titles
-        String s2= ".*(?i)"+title+".*"; // regex for searching the word start anywhere in the title
+        String startsWithTitle = "\\b(?i)(" + title + ")\\b"; // regex for matching search at start of title
+        String containsTitle = ".*(?i)" + title + ".*"; // regex for matching search anywhere in title
+
+        // Adds all exact matches to matchList
         for (Movie movie : ALL_MOVIES)
         {
-            if (movie.getTitle().matches(s1) || movie.getTitle().matches(s2))
-            {
-                matches.add(movie);
-
-            }
-            if (movie.getTitle().matches(s1))
+            if (movie.getTitle().matches(startsWithTitle))
             {
-                m = movie;
-                for (Movie match : matches)
-                {
-                    if (s.contains(match))
-                    {
-                        m = matches.get(0);
-                    }
-                    else s.add(match);
-                }
+                matchList.add(movie);
             }
-            else if (movie.getTitle().matches(s2) && m==null)
+        }
+
+        // Adds all movies that contain the title searched to the end of matchList
+        for (Movie movie : ALL_MOVIES)
+        {
+            if (movie.getTitle().matches(containsTitle) && !matchList.contains(movie))
             {
-                m = movie;
+                matchList.add(movie);
             }
         }
+
+        // When year is specified, searches matchList for title with matching year, returning first match
         if (!"none".equals(year))
         {
-            for (Movie movie : matches)
+            for (Movie movie : matchList)
             {
-                if (movie.getYear().equalsIgnoreCase(year)) m = movie;
+                if (movie.getYear().equalsIgnoreCase(year))
+                {
+                    m = movie;
+                    break;
+                }
             }
         }
+        else // returns first entry in matchList when year is not specified
+        {
+            m = matchList.get(0);
+        }
+
         if (m == null)
         {
             throw new MovieNotFoundException();
```

Comment:
I believe this line is never being hit because
```
http://localhost:8080/movie/asdf
```
returns
```JSON
{
  "statusCode": 404,
  "error": "Index 0 out of bounds for length 0"
}
```